opossum-tool/OpossumUI
# fix: Enable context isolation

### Summary of changes

- Activate context isolation
- Use a context bridge to expose the `ipcRenderer`
- Instead of exposing the full `ipcRenderer` API a subset of functions is exposed
    - for the ocassions where `.invoke` is called an API is defined, which describes the actual action, e.g. `openLink`
    - when listeners are used (`.on` and `.removeListener`) the respective part of the API is exposed but an allow list of channels is applied in the backend before passing anything through  

### Context and reason for change

Deactivating context isolation and exposing `ipcRenderer` globally is discouraged. An attacker could access it from the frontend with `window.ipcRenderer` and would have a rather powerful API. Instead it is recommended to use a preload script where a subset of APIs is exposed.
Details can be found [here](https://www.electronjs.org/docs/latest/tutorial/context-isolation).

### How can the changes be tested

- run all tests
- more importantly open the app and check the actions invoking the `ipcRenderer` are working (saving files, opening links, ...)
